src/vs/workbench/contrib/debug/common/replModel.ts

```diff
@@ -174,13 +174,18 @@ export class ReplGroup implements IReplElement {
 	}
 }
 
+type FilterFunc = ((element: IReplElement) => void);
+
 export class ReplModel {
 	private replElements: IReplElement[] = [];
 	private readonly _onDidChangeElements = new Emitter<void>();
 	readonly onDidChangeElements = this._onDidChangeElements.event;
+	private filterFunc: FilterFunc | undefined;
 
 	getReplElements(): IReplElement[] {
-		return this.replElements;
+		return this.replElements.filter(element =>
```

Comment:
Is this really needed? Do the problems do the same?
I thought the tree will do the filtering even if we give it more data? Or I might be mistaken?

Reply:
unfortunately, this model is literally a model of real data source which has nothing to do with actual implementation (ReplDataSource). I didn't found better way to test this than to add support of primitive filtering to the model.